From 5278c89163fc3600b008cf8765a16dc48f0c6172 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Wed, 13 Dec 2017 14:57:06 +0800
Subject: [PATCH 24/27] ueventd: fix the buggy logging

LOG(xxx) is totally broken. Use KLOG_xxx to actually show the log.
---
 init/ueventd.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/init/ueventd.cpp b/init/ueventd.cpp
index a284203ba1..ee605f433f 100644
--- a/init/ueventd.cpp
+++ b/init/ueventd.cpp
@@ -31,6 +31,7 @@
 #include <android-base/logging.h>
 #include <android-base/properties.h>
 #include <fstab/fstab.h>
+#include <cutils/klog.h>
 #include <selinux/android.h>
 #include <selinux/selinux.h>
 
@@ -212,7 +213,8 @@ void ColdBoot::Run() {
     WaitForSubProcesses();
 
     close(open(COLDBOOT_DONE, O_WRONLY | O_CREAT | O_CLOEXEC, 0000));
-    LOG(INFO) << "Coldboot took " << cold_boot_timer.duration().count() / 1000.0f << " seconds";
+    klog_set_level(KLOG_NOTICE_LEVEL);
+    KLOG_NOTICE("Coldboot", "took %f seconds", cold_boot_timer.duration().count() / 1000.0f);
 }
 
 DeviceHandler CreateDeviceHandler() {
-- 
2.17.1

